packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -315,7 +315,11 @@ export class InterfaceVpcEndpointAwsService implements IInterfaceVpcEndpointServ
   public readonly privateDnsDefault?: boolean = true;
 
   constructor(name: string, prefix?: string, port?: number) {
-    this.name = `${prefix || 'com.amazonaws'}.${Aws.REGION}.${name}`;
+    // this.name = `${prefix || 'com.amazonaws'}.${Aws.REGION}.${name}`;
+    const region = Lazy.stringValue({
```

Comment:
This doesn't need to be lazy. Could just be `Stack.of(context.scope).region`, has the same behavior.

Reply:
I'm not sure what the code should be here. Here's what I've tried, all without success:

```ts
    const region = Stack.of(context.scope).region;
```

Yields "cannot find name 'context'"

```ts
    const region = {
      produce: (context: any) => Stack.of(context.scope).region,
    };
``` 
Results in a test failure because `Stack.of(context.scope).region` resolves to an Object:

```
Field ServiceName mismatch: com.amazonaws.[object Object].execute-api !== com.amazonaws.us-east-1.execute-api,Field SubnetIds mismatch: element 1: Ref: VPCPrivateSubnet2SubnetCFCDAA7A !== VPCPrivateSubnet3Subnet3EDCD457 in:
```

Reply:
InterfaceVpcEndpointAwsService isn't a construct you don't have any context/scope/stack to use to get the region from. Using Lazy lets you get that context from wherever the InterfaceVpcEndpointAwsService is used.

Reply:
I see. You are correct good sir, my apologies.